libs/libcommon/src/libcommon/viewer_utils/asset.py

```diff
@@ -24,6 +27,30 @@ def create_asset_dir(
     return dir_path, url_dir_path
 
 
+def glob_rows_in_assets_dir(
+    dataset: str,
+    assets_directory: StrPath,
+) -> Generator[Path, None, None]:
+    return Path(assets_directory).resolve().glob(os.path.join(dataset, DATASET_SEPARATOR, "*", "*", "*"))
+
+
+def update_last_modified_date_of_rows_in_assets_dir(
+    dataset: str,
+    config: str,
+    split: str,
+    offset: int,
+    length: int,
+    assets_directory: StrPath,
+) -> None:
+    row_dirs_path = Path(assets_directory).resolve() / dataset / DATASET_SEPARATOR / config / split
+    for row_idx in range(offset, offset + length):
+        if (row_dirs_path / str(row_idx)).is_dir():
+            # update the directory's last modified date
+            if (row_dirs_path / str(row_idx) / DATASETS_SERVER_MDATE_FILENAME).is_file():
+                (row_dirs_path / str(row_idx) / DATASETS_SERVER_MDATE_FILENAME).unlink()
+            (row_dirs_path / str(row_idx) / DATASETS_SERVER_MDATE_FILENAME).touch()
```

Comment:
is it necessary to delete the file if it exists? afaiu, touch is meant to update the modification date

https://docs.python.org/3/library/pathlib.html#pathlib.Path.touch

> If the file already exists, the function succeeds if exist_ok is true (and its modification time is updated to the current time), otherwise [FileExistsError](https://docs.python.org/3/library/exceptions.html#FileExistsError) is raised.

Reply:
`touch` is not enough to modify the last modified time of **the parent directory** when the file already exists.

Reply:
hmmm ok